lib/util/progress.coffee

```diff
@@ -1,53 +1,238 @@
-# TODO: move determinate progress bars to the front
-# TODO: hover UI with progress stack, descriptions
+{Emitter} = require 'atom'
+
+# Progress Bars
+#
+# This module provides an API for displaying progress bars in Atom. The methods
+# below allow modifing the stack of progress bars, which is represented by
+# corresponding UI elements:
+#    - A "global" progress bar is shown in the status bar. In pratice, this is
+#      the first determinate progress bar in the stack. If there is none, the
+#      first element (which then is indeterminate) will be shown instead. If the
+#      stack is empty, an empty progress bar will be shown instead.
```

Comment:
You may have addressed this, but we should make sure to handle the special case where the bottom progress bar is at 0 – and then the global bar should be indeterminate.